nixos/modules/services/misc/gitea.nix

```diff
@@ -15,6 +15,7 @@ let
     APP_NAME = ${cfg.appName}
     RUN_USER = ${cfg.user}
     RUN_MODE = prod
+    WORK_PATH = ${cfg.stateDir}
```

Comment:
I was wondering if it wouldn't be worth pulling up the `extraConfig` line (`${optionalString (cfg.extraConfig != null) cfg.extraConfig}`) to before the main `settings` section.

That way, a non-sectioned setting like `WORK_PATH` could be added via `extraConfig` as well. Right now such a setting would get assigned randomly to whatever the last `settings` section turned out to be, right?

Do you see any trade-offs here, @emilylange?

Reply:
From https://github.com/go-gitea/gitea/pull/25330

> The "work path" priority is: WORK_PATH in app.ini > cmd arg --work-path > env var GITEA_WORK_DIR > builtin default

We seem to be doing something similar with `GITEA_WORK_DIR` at
https://github.com/NixOS/nixpkgs/blob/687ed41b532af50e440258159b4360e783bad87e/nixos/modules/services/misc/gitea.nix#L625
and
https://github.com/NixOS/nixpkgs/blob/687ed41b532af50e440258159b4360e783bad87e/nixos/modules/services/misc/gitea.nix#L665

Assuming the env var does not try to update the specified `app.ini`, this might be a more suitable place.
Or, alternatively, a `lib.mkDefault`ed `WORK_PATH = cfg.stateDir` in `config` similar to https://github.com/NixOS/nixpkgs/blob/687ed41b532af50e440258159b4360e783bad87e/nixos/modules/services/misc/gitea.nix#L391-L392

I don't know why one would want to manually override `WORK_PATH` when using the module, but ehh, maybe there are use-cases.
And both solutions would allow for that :)

I don't see much value changing the order of the freeform `settings` option and `extraConfig`.
But maybe I just don't quite get what you mean with that^^

Reply:
I don't really know what's going on either. If there is no `WORK_PATH` in `app.ini`, it will try to add it (possibly from that env var you're mentioning). That will fail when app.ini is read-only.

Maybe that could be considered a bug upstream, but I wanted to leave this here, so that we already know when doing the update.

Reply:
> I don't see much value changing the order of the freeform settings option and extraConfig.
> But maybe I just don't quite get what you mean with that^^

What I meant is: I couldn't add the `WORK_PATH` through extra config, because:

```
UNSECTIONED_CONFIG1 = value1

[section1]
SECTION_CONFIG = value2

EXTRA_CONFIG = value3
```

here, `EXTRA_CONFIG = value3` would be considered part of `[section1]`, even though it's meant to be unsectioned. There is also no code that could reasonably rely on this behavior, since the order of sections is essentially random and so the section that an unsectioned `extraConfig` is assigned to would also be random.

Reply:
TLDR but extraConfig is deprecated and we should remove it rather sooner than later.

Reply:
Good to know.

Do our `.ini` config helpers do the unsectioned preamble?